components/enrichers/reachability/internal/atom/reader.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/ocurity/dracon/components/enrichers/reachability/internal/logging"
+)
+
+type (
+	// Reader is responsible for managing how to read atom files and understand their contents.
+	Reader struct {
+		atomFilePath string
+		purlParser   *purl.Parser
+	}
+
+	// ReachablePurls maps reachable purls based on the report.
+	ReachablePurls map[string]struct{}
+
+	// Reachables is a slice of Reachable.
+	Reachables []Reachable
+
+	// Response maps the content of an atom reachability report in json format.
+	Response struct {
+		Reachables Reachables `json:"reachables"`
+	}
+
+	// Reachable represents an atom reachable result.
+	Reachable struct {
+		Flows []Flows  `json:"flows"`
+		Purls []string `json:"purls"`
+	}
+
+	// Flows describes the flows on how to reach such reachable.
+	Flows struct {
+		ID                    int    `json:"id"`
+		Label                 string `json:"label"`
+		Name                  string `json:"name"`
+		FullName              string `json:"fullName"`
+		Signature             string `json:"signature"`
+		IsExternal            bool   `json:"isExternal"`
+		Code                  string `json:"code"`
+		TypeFullName          string `json:"typeFullName"`
+		ParentMethodName      string `json:"parentMethodName"`
+		ParentMethodSignature string `json:"parentMethodSignature"`
+		ParentFileName        string `json:"parentFileName"`
+		ParentPackageName     string `json:"parentPackageName"`
+		ParentClassName       string `json:"parentClassName"`
+		LineNumber            int    `json:"lineNumber"`
+		ColumnNumber          int    `json:"columnNumber"`
+		Tags                  string `json:"tags"`
+	}
+)
+
+// NewReader returns a new atom file reader.
+func NewReader(atomFilePath string, purlParser *purl.Parser) (*Reader, error) {
+	switch {
+	case atomFilePath == "":
+		return nil, errors.New("invalid empty atom file path")
+	case purlParser == nil:
+		return nil, errors.New("invalid nil purl parser")
+	}
+
+	return &Reader{
+		atomFilePath: atomFilePath,
+		purlParser:   purlParser,
+	}, nil
+}
+
+// Read deserialises the json content of the provided atom file into Reachables format.
+func (r *Reader) Read() (*Response, error) {
+	b, err := os.ReadFile(r.atomFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read atom file: %w", err)
+	}
+
+	var res Response
+	if err := json.Unmarshal(b, &res); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal atom response: %w", err)
+	}
+
+	return &res, nil
+}
+
+// ReachablePurls finds all the reachable purls presents in the atom reachability result.
+func (r *Reader) ReachablePurls(ctx context.Context, reachables *Response) (ReachablePurls, error) {
+	logger := logging.FromContext(ctx)
+
+	rawPurls, err := jmespath.Search("reachables[].purls[]", reachables)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search reachable purls: %w", err)
+	}
+
+	purls, ok := rawPurls.([]any)
+	if !ok {
+		logger.Error(
```

Comment:
can we please replace with slog? we are trying to switch components to log in a structured way